dubbo-registry/dubbo-registry-zookeeper/src/main/java/org/apache/dubbo/registry/zookeeper/ZookeeperServiceDiscoveryChangeWatcher.java

```diff
@@ -68,7 +63,7 @@ public ZookeeperServiceDiscoveryChangeWatcher(ZookeeperServiceDiscovery zookeepe
         this.notifier = new RegistryNotifier(zookeeperServiceDiscovery.getDelay(), getExecutorRepository().getServiceDiscoveryAddressNotificationExecutor()) {
             @Override
             protected void doNotify(Object rawAddresses) {
-                listeners.forEach(listener -> listener.onEvent((ServiceInstancesChangedEvent)rawAddresses));
+                zookeeperServiceDiscovery.dispatchServiceInstancesChangedEvent((ServiceInstancesChangedEvent)rawAddresses);
```

Comment:
Can listeners in Watcher being replaced with listeners in ServiceDiscovery?

Watcher is bound to one single service while ServiceDiscovery is shared among services.